tokio/tests/sync_watch.rs

```diff
@@ -368,3 +368,52 @@ async fn receiver_is_notified_when_last_sender_is_dropped() {
 
     assert!(t.is_woken());
 }
+
+#[tokio::test]
+async fn receiver_changed_is_cooperative() {
+    let (tx, mut rx) = watch::channel(());
+
+    drop(tx);
+
+    tokio::select! {
+        biased;
+        _ = async {
+            loop {
+                let _ = rx.changed().await;
```

Comment:
You may need to send another message for this test to actually work? Did you verify that these tests fail without the change?

Reply:
I ran the test without the fix of this PR locally and it was not failing but ran forever. 
I then took a look at the other tests for cooperativeness and found that they also just hang in that case so I thought that this would be fine.

See this from `tokio/tests/io_repeat.rs` for example:
```
#[tokio::test]
async fn repeat_poll_read_is_cooperative() {
    tokio::select! {
        biased;
        _ = async {
            loop {
                let mut buf = [0u8; 4096];
                tokio::io::repeat(0b101).read_exact(&mut buf).await.unwrap();
            }
        } => {},
        _ = tokio::task::yield_now() => {}
    }
}

```

Reply:
Yes I ran the test without the cooperativeness and found that it would just loop forever. I than checked other tests for cooperativeness like the one in `tokio/tests/io_repeat.rs` and found that they would also just hang without the cooperativeness so I thought this would be ok.